packages/hint/src/lib/engine.ts

```diff
@@ -300,13 +319,21 @@ export class Engine extends EventEmitter {
         return this.connector.querySelectorAll(selector);
     }
 
-    public emitAsync(event: string | Array<string>, ...values: Array<any>): Promise<Array<any>> {
+    public emit<K extends StringKeyOf<E>>(event: K, data: E[K]): boolean {
+        return super.emit(event, data);
+    }
+
+    public emitAsync<K extends StringKeyOf<E>>(event: K, data: E[K]): Promise<any[]> {
```

Comment:
We are using the form `Array<>` everywhere.
Maybe it is a good moment to change to `type[]` everywhere if everybody like that form the most :)

Reply:
I know both @molant and I prefer `type[]`. I should be able to (carefully) do this with a multi-file search-and-replace (as there are over 500 references that still use `Array<type>` in the project and I'd need to watch out for nested generics).

@webhintio/contributors Any objections?

Reply:
go for it! :)

Reply:
No objections. Let's do it!